tests/integration/search_test.t

```diff
@@ -0,0 +1,178 @@
+#!/usr/bin/perl -w
+
+use ProductOpener::PerlStandards;
+
+use Test::More;
+use ProductOpener::APITest qw/:all/;
+use ProductOpener::Test qw/:all/;
+use LWP::Simple "get";
+use Data::Dumper;
+use Encode;
+use File::Basename "dirname";
+use JSON::PP;
+use ProductOpener::TestDefaults qw/:all/;
+use ProductOpener::Test qw/:all/;
+use Getopt::Long;
+
+my $test_name = "search_test";
+my $tests_dir = dirname(__FILE__);
+my $expected_dir = $tests_dir . "/expected_test_results/" . $test_name;
+
+my @products = (
+	{
+		code => '200000000034',
+		lang => "en",
+		product_name => "test_1",
+		generic_name => "Tester",
+		quantity => "100 ml",
+		link => "https://github.com/openfoodfacts/openfoodfacts-server",
+		expiration_date => "test",
+		ingredients_text => "apple, milk, eggs, palm oil",
+		origin => "france",
+		serving_size => "10g",
+		packaging_text => "no",
+		action => "process",
+		type => "add",
+		".submit" => "submit"
+	},
+	{
+		code => '200000000037',
+		lang => "en",
+		product_name => "vegan & palm oil free",
+		generic_name => "Tester",
+		quantity => "100 ml",
+		link => "https://github.com/openfoodfacts/openfoodfacts-server",
+		expiration_date => "test",
+		ingredients_text => "fruit, rice",
+		origin => "france",
+		serving_size => "10g",
+		packaging_text => "no",
+		action => "process",
+		type => "add",
+		".submit" => "submit"
+
+	},
+	{
+		code => '200000000038',
+		lang => "en",
+		product_name => "palm oil free & non vegan",
+		generic_name => "Tester",
+		quantity => "100 ml",
+		link => "https://github.com/openfoodfacts/openfoodfacts-server",
+		expiration_date => "test",
+		ingredients_text => "apple, milk, eggs",
+		origin => "france",
+		serving_size => "10g",
+		packaging_text => "no",
+		action => "process",
+		type => "add",
+		".submit" => "submit"
+	},
+	{
+		code => '200000000039',
+		lang => "es",
+		product_name => "Vegan Test Snack",
+		generic_name => "Tester",
+		quantity => "100 ml",
+		link => "https://github.com/openfoodfacts/openfoodfacts-server",
+		expiration_date => "test",
+		ingredients_text => "apple, water, palm oil",
+		origin => "spain",
+		serving_size => "5g",
+		packaging_text => "no",
+		action => "process",
+		categories => "snacks",
+		type => "add",
+		".submit" => "submit"
+	},
+	{
+		code => '200000000045',
+		lang => "es",
+		product_name => "Vegan Test Snack",
+		generic_name => "Tester",
+		quantity => "100 ml",
+		link => "https://github.com/openfoodfacts/openfoodfacts-server",
+		expiration_date => "test",
+		ingredients_text => "apple, water",
+		origin => "France",
+		serving_size => "5g",
+		packaging_text => "no",
+		action => "process",
+		categories => "breakfast cereals",
+		type => "add",
+		".submit" => "submit"
+	}
+);
```

Comment:
you did not take into account [this comment](https://github.com/openfoodfacts/openfoodfacts-server/pull/7170/files/aafebe3ee9c499ddbd42aaafdc86d871bfd1c34c#r962885023)

You should have a %default_product (in TestDefault.pm) and use it here to avoid having big json.

Reply:
@alexgarel okay :)